cardano-testnet/src/Testnet/Property/Run.hs

```diff
@@ -22,43 +25,72 @@ import qualified System.Console.ANSI as ANSI
 import           System.Console.ANSI (Color (..), ColorIntensity (..), ConsoleLayer (..), SGR (..))
 import           System.Directory
 import qualified System.Exit as IO
+import           System.FilePath ((</>), normalise)
 import qualified System.Info as SYS
 import qualified System.IO as IO
+import           Text.Printf (printf)
+
+import           Cardano.Api.IO (unFile)
 
 import           Testnet.Property.Util (integration, integrationWorkspace)
-import           Testnet.Start.Types
+import           Testnet.Start.Types (UserProvidedEnv (..), Conf, mkConf)
+import           Testnet.Types (TestnetNode (..), TestnetRuntime (..), spoNodes)
 
 import           Hedgehog (Property)
 import qualified Hedgehog as H
+import           Hedgehog.Extras.Stock.IO.Network.Sprocket (Sprocket (..))
 import           Hedgehog.Extras.Stock.OS (isWin32)
 import qualified Hedgehog.Extras.Test.Base as H
 import           Test.Tasty.ExpectedFailure (wrapTest)
 import qualified Test.Tasty.Hedgehog as H
 import           Test.Tasty.Providers (testPassed)
 import           Test.Tasty.Runners (Result (resultShortDescription), TestTree)
 
-runTestnet :: UserProvidedEnv -> (Conf -> H.Integration a) -> IO ()
+runTestnet :: UserProvidedEnv -> (Conf -> H.Integration TestnetRuntime) -> IO ()
 runTestnet env tn = do
-  tvRunning <- STM.newTVarIO False
+  tvRunning <- STM.newTVarIO Nothing
 
   void . H.check $ testnetProperty env $ \c -> do
-    void $ tn c
-    H.evalIO . STM.atomically $ STM.writeTVar tvRunning True
-
-  running <- STM.readTVarIO tvRunning
-
-  if running
-    then do
+    runtime <- tn c
+    H.evalIO . STM.atomically $ STM.writeTVar tvRunning (Just runtime)
+
+  STM.readTVarIO tvRunning >>= \case
+    Just runtime@TestnetRuntime
+      { configurationFile
+      , testnetMagic
+      } -> do
       ANSI.setSGR [SetColor Foreground Vivid Green]
-      IO.putStr "Testnet is running.  Type CTRL-C to exit."
+      IO.putStrLn $ printf
+        "Please disregard the message above implying a failure.\n\
```

Comment:
This is why we shouldn't be using properties. They only ought to be used in property testing.

Reply:
I think you mean `PropertyT IO` as a base monad for testnet setup?

I think I saw someone complaining about `cardano-testnet` exeuctable console output breaking some tooling because it was detecting colours incorrectly.